pkg/kubelet/oom/oom_watcher_linux.go

```diff
@@ -59,25 +65,38 @@ func NewWatcher(recorder record.EventRecorder) (Watcher, error) {
 }
 
 const (
-	systemOOMEvent           = "SystemOOM"
-	recordEventContainerName = "/"
+	systemOOMEvent                 = "SystemOOM"
+	cGroupOOMEvent                 = "CgroupOOM"
+	legacyRecordEventContainerName = "/"
 )
 
 // Start watches for system oom's and records an event for every system oom encountered.
-func (ow *realWatcher) Start(ref *v1.ObjectReference) error {
+func (ow *realWatcher) Start(ref *v1.ObjectReference, getPodByUID GetPodByUIDFunc) error {
 	outStream := make(chan *oomparser.OomInstance, 10)
 	go ow.oomStreamer.StreamOoms(outStream)
 
 	go func() {
 		defer runtime.HandleCrash()
 
 		for event := range outStream {
-			if event.VictimContainerName == recordEventContainerName {
-				klog.V(1).InfoS("Got sys oom event", "event", event)
-				eventMsg := "System OOM encountered"
-				if event.ProcessName != "" && event.Pid != 0 {
-					eventMsg = fmt.Sprintf("%s, victim process: %s, pid: %d", eventMsg, event.ProcessName, event.Pid)
+			eventMsg := "System OOM encountered"
+			if event.ContainerName != legacyRecordEventContainerName {
+				eventMsg = "Cgroup OOM encountered"
+			}
+			if event.ProcessName != "" && event.Pid != 0 {
+				eventMsg = fmt.Sprintf("%s, victim process: %s, pid: %d", eventMsg, event.ProcessName, event.Pid)
+			}
+
+			parsedContainer := containerRegexp.FindStringSubmatch(event.VictimContainerName)
+			if parsedContainer != nil && event.ContainerName != legacyRecordEventContainerName {
+				klog.V(1).InfoS("Got cgroup oom event", "event", event)
+				if pod, ok := getPodByUID(types.UID(parsedContainer[1])); ok {
```

Comment:
parsedContainer at index 1 is not guaranteed to be set with FindStringSubmatch. We need a check on line 91 to make sure it exists.

Reply:
Good catch, I've fixed this in the latest commit.